src/components/map/geometry-item.tsx

```diff
@@ -129,9 +113,14 @@ export const GeometryItem = ({
                     weight: 10,
                     opacity: 0,
                   }}
-                  className={layer.type}
+                  eventHandlers={{
+                    mouseover: e => e.target.setStyle({ opacity: 0.5 }),
+                    mouseout: e => e.target.setStyle({ opacity: 0 }),
+                    ...featureEventHandler,
+                  }}
+                  className={`streams-hover ${pointerClassName}`}
```

Comment:
Maybe you should use `cn` helper to concat and merge classes to get something like this `className={cn('streams-hover', !hasDetails && '!cursor-[unset]')}`